File: client/python/ping.py
#!/usr/bin/env python3
"""RSP ping tool — Python equivalent of client/nodejs/ping.js."""

import argparse
import asyncio
import json
import sys

from rsp_client import RSPClient


async def main() -> None:
    parser = argparse.ArgumentParser(description="Ping an RSP node")
    parser.add_argument("transport_spec", help="tcp:<host>:<port>")
    parser.add_argument("destination_node_id", help="target node UUID")
    args = parser.parse_args()

    async with RSPClient() as client:
        await client.connect(args.transport_spec)
        ok = await client.ping(args.destination_node_id)
        print(json.dumps({
            "local_node_id": client.node_id,
            "resource_manager_node_id": client.peer_node_id,
            "destination_node_id": args.destination_node_id,
            "success": ok,
        }, indent=2))
        if not ok:
            sys.exit(1)


if __name__ == "__main__":
    asyncio.run(main())

File: client/nodejs_full/ping.js
'use strict';

const path = require('path');
const {RSPClient} = require('./rsp_client');

async function main() {
    const [, , transportSpec, destinationNodeId] = process.argv;
    if (!transportSpec || !destinationNodeId) {
        console.error('Usage: node client/nodejs_full/ping.js tcp:<host>:<port> <destination-node-id>');
        process.exit(1);
    }

    // Locate the repo root so messages.proto can be loaded into the registry for
    // callers that switch to protobuf encoding.  The ping itself uses JSON encoding
    // by default (no registry use required).
    const protoRoot = path.resolve(__dirname, '..', '..');

    const client = new RSPClient(null, {protoRoot});
    try {
        await client.connect(transportSpec);
        const ok = await client.ping(destinationNodeId);
        console.log(JSON.stringify({
            local_node_id: client.nodeId,
            resource_manager_node_id: client.peerNodeId,
            destination_node_id: destinationNodeId,
            success: ok,
        }, null, 2));
        if (!ok) process.exit(1);
    } finally {
        await client.close().catch(() => {});
    }
}

main().catch((error) => {
    console.error(`nodejs_full ping failed: ${error.message}`);
    process.exit(1);
});
